nf_core/pipeline-template/.devcontainer/Dockerfile

```diff
@@ -0,0 +1,21 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/python-3-anaconda/.devcontainer/base.Dockerfile
+
+FROM mcr.microsoft.com/vscode/devcontainers/anaconda:0-3
```

Comment:
This is the same for all pipelines, right? Can we not have this at the nf-core/tools level and then just pull into codespaces? Instead of building the image every time?

Would also be nice to see if we could combine with [the image](https://github.com/nf-core/tools/blob/dev/nf_core/gitpod/gitpod.Dockerfile) that we have for GitPod already as they're basically serving the same purpose I think. Curious to see if @mahesh-panchal has any thoughts on this.

Looking at the gitpod image I guess that's going to be too specific (unless we tweak settings to be able to use the same username etc), but it could be nice to at least keep the two recipes in the same directory etc.

Reply:
If I'm not misunderstanding, I think you can just pull the Gitpod image and use that. This image definition installs conda, nextflow, and nf-core. The Gitpod image has that as well as docker ( in docker ), mamba, prettier, etc. I think they should both be the same.

Reply:
This also has vscode, jupyter notebooks as well

Reply:
Does that mean the codespaces container needs to provide all interfaces too? The Gitpod environment uses VScode, but doesn't include it in the container ( and I think the Gitpod user can select other IDE too if they prefer ).

Reply:
Actually I might be wrong about VS code not being in the container. I'm not really sure what's in there besides docker

Reply:
> I did more digging around and there should be some way to combine these: https://github.com/gitpod-io/template-gitpod-sxs-codespaces. The Codespaces environment is using the Gitpod image here.
> 
> The Gitpod base image does have a lot of things we generally don't need, but it was used for docker and java included ( full list is https://github.com/gitpod-io/workspace-images/blob/HEAD/dazzle.yaml#L3. I was wondering initially if the Gitpod image could use the devcontainers image as a base, but I can't find what's included in them.
> 
> Also what's the difference between Codespaces and Gitpod ? Gitpod has it's own comparison https://www.gitpod.io/vs/github-codespaces. but I haven't found a simple independent review



> I did more digging around and there should be some way to combine these: https://github.com/gitpod-io/template-gitpod-sxs-codespaces. The Codespaces environment is using the Gitpod image here.
> 
> The Gitpod base image does have a lot of things we generally don't need, but it was used for docker and java included ( full list is https://github.com/gitpod-io/workspace-images/blob/HEAD/dazzle.yaml#L3. I was wondering initially if the Gitpod image could use the devcontainers image as a base, but I can't find what's included in them.
> 
> Also what's the difference between Codespaces and Gitpod ? Gitpod has it's own comparison https://www.gitpod.io/vs/github-codespaces. but I haven't found a simple independent review

Thanks will dig deeper to see if I can syncronize the two.